src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -395,26 +395,57 @@ struct blk_zone_range make_range(
   };
 }
 
-using blk_open_zone_ertr = crimson::errorator<
+using blk_zone_op_ertr = crimson::errorator<
   crimson::ct_error::input_output_error>;
-using blk_open_zone_ret = blk_open_zone_ertr::future<>;
-blk_open_zone_ret blk_open_zone(seastar::file &device, blk_zone_range &range){
+using blk_zone_op_ret = blk_zone_op_ertr::future<>;
+blk_zone_op_ret blk_zone_op(seastar::file &device,
+		            blk_zone_range &range,
+			    zone_op_t zone_op){
+  LOG_PREFIX(ZNSSegmentManager::blk_zone_op);
+
+  unsigned long ioctl_op = 0;
+  std::string ioctl_name = "";
+  switch (zone_op) {
+    case zone_op_t::ZONE_OPEN:
+      ioctl_op = BLKOPENZONE;
+      ioctl_name = "BLKOPENZONE";
```

Comment:
better off specialize a `fmt::formatter<zone_op_t>` . so we don't need to 1) create a `std::string` , 2) call its assignment operator and then 3) capture it by value, only for the failure case.

Reply:
Thanks for the review.
Added fmt::formatter<zone_op_t>.